### DISCUSSION TITLE
Konversi integer ke datetime (hour-minute)

### DISCUSSION THREAD

**Discussion by fxpeterbs31** [teamalgoritma/community#312] [Q&A] [answered]
Halo, saya sedang mengerjakan case capstone ML dan menemukan masalah pada pengubahan integer menjadi datetime. 
Awalnya saya sudah berhasil mengubah dari integer ke string dengan format date (tapi masih bertipe data string). Tetapi ketika mengubah string tersebut menjadi datetime, saya menemukan masalah yaitu: 
1. Ketika memakai base, saya malah mengubahnya ke format lengkap (ymd_hms)
<img width="62" alt="2022-09-20 22_32_59-Capstone_ML - RStudio" src="https://user-images.githubusercontent.com/107398327/191301222-4951076f-c6d2-4900-aa5c-70b30e44d163.png">

2. Ketika menggunakan lubridate, malah hasilnya period
<img width="66" alt="2022-09-20 22_22_37-Capstone_ML - RStudio" src="https://user-images.githubusercontent.com/107398327/191300410-4fc5a317-e55f-4010-8d77-f9e789e635dd.png">

Apakah data ini bisa untuk diubah menjadi time, contoh: 320 menjadi 3:20 (tipe data datetime)? Lalu pada capstone ml ini, sebenarnya apakah perlu untuk diubah kedalam bentuk datetime? atau bentuk character saja cukup?? 

Terima kasih

**Comment by NabiilahArdini** [comment, marked as answer]
Dear @fxpeterbs31, 

Hasil dari ekstraksi informasi hour-minute dari lubridate memang berupa periode.
Untuk perlukah data tersebut diubah ke format hour-minute (eg. 3:20) silahakan dipertimabngkan kembali, karena bila begitu, maka nilai prediktornya akan berupa karakter/periode unik "3:20" saja. Bila data test memiliki nilai "3:21" maka tidak dapat disamakan dengan data trainingnya.
Opsi lain, apabila Bapak ingin memecah nilai 320 menjadi jam 3 dan menit 20 di kolom terpisah, Bapak bisa gunakan fungsi `separate() `dari dplyr. Berikut adalah dokumentasinya: https://tidyr.tidyverse.org/reference/separate.html

Semoga membantu!
